setup.py

```diff
@@ -51,7 +51,10 @@ def get_long_desc():
     package_dir={'': 'src'},
     include_package_data=True,
     extras_require={
-        'pandas': ['pandas'],
+        'pandas:python_version=="3.4"': ['pandas~=0.19.0'],
+        'pandas:python_version=="2.7" or python_version>="3.5"': ['pandas'],
+        'testing:python_version=="3.4"': ['pandas~=0.19.0'],
```

Comment:
I wish I could use environment markers in the `tests_requires` list but apparently this is not supported, so here's a work-around.